### PR TITLE
Added support for Tessera

### PR DESCRIPTION
**Pull request by ksivane**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -114,11 +116,18 @@ RUN apt-get update -qq \
...
+RUN apt-get install -y openjdk-8-jdk && \
+	apt-get install -y ant
+
 # Install constellation
 RUN wget -q https://github.com/jpmorganchase/constellation/releases/download/v${CONSTELLATION_VERSION}-build.1/constellation-${CONSTELLATION_VERSION}-ubuntu1604.tar.gz  -O - \
     | tar xzO constellation-node > /usr/local/bin/constellation-node \
     && chmod +x /usr/local/bin/constellation-node
 
+# Install tessera
+RUN wget https://oss.sonatype.org/content/groups/public/com/jpmorgan/quorum/tessera-app/0.9.2/tessera-app-${TESSERA_VERSION}-app.jar -O ${TESSERA_LOC}/tessera
```
Comment (easeev) [inline Dockerfile]: Why fixed version `0.9.2`?

**Comment by ksivane** [comment]
@easeev This PR is just for the Quorum docker image....it installs tessera binary into the image. That's all.
`start_quorum_node.sh` will have updates along with the quorum-toolbox.

You can still use this updated quorum image....it will not affect anything.

**Comment by easeev** [comment]
> @easeev This PR is just for the Quorum docker image....it installs tessera binary into the image. That's all.

Ok, probably PR needs a better description then :)

**Comment by hayorov** [comment]
@ksivane Please create a separate docker image for tessera service, and take into account that it will be 2 separate service: geth daemon and tessera.

**Comment by ksivane** [comment]
> @ksivane Please create a separate docker image for tessera service, and take into account that it will be 2 separate service: geth daemon and tessera.

Creating separate Tessera image is ok. But how will geth talk to the tessera process? The Quorum To Tessera [Api](https://github.com/jpmorganchase/tessera/wiki/Interface-&-API) is private and happens over a file system ipc file.  So, you intend to have a somethng like a shared ipc file between the Geth and Tessera container?

**Comment by hayorov** [comment]
> > @ksivane Please create a separate docker image for tessera service, and take into account that it will be 2 separate service: geth daemon and tessera.
> 
> Creating separate Tessera image is ok. But how will geth talk to the tessera process? The Quorum To Tessera [Api](https://github.com/jpmorganchase/tessera/wiki/Interface-&-API) is private and happens over a file system ipc file. So, you intend to have a somethng like a shared ipc file between the Geth and Tessera container?

For Q2T you can use INET (Rest) instead of socket:
```
  "serverConfigs": [
    {
      "app": "Q2T",
      "enabled": true,
      "serverSocket": {
        "type": "INET",
        "port": 9000,
        "hostName": "http://0.0.0.0"
      },
      "communicationType": "REST"
    },
```

relay on internal k8s DNS to point the geth - you can find example in quorum-explorer

**Comment by ksivane** [comment]
@hayorov 

You are right that Tessera Q2T can be configured for REST.  But how to configure Geth to talk to Tessera over REST? All examples and documentations i am seeing so far launch geth like this:

`PRIVATE_CONFIG=/path/to/ipc geth --datadir new-node-1 .........`

[This](http://docs.goquorum.com/en/latest/Getting%20Started/Getting-Started-From-Scratch/) says
`Tessera IPC bridge will be over a file name defined in your config.json, usually named tm.ipc as evidenced in prefix PRIVATE_CONFIG=tm.ipc.`

Am i missing something here, can you point me to some example/doc where Geth and Tessera are both configured to talk to each other over REST?

**Comment by ksivane** [comment]
@hayorov @easeev 

Consider Tessera related work as done. Plz note the following:

- All work done on Tessera `v0.9.2`
- Tessera and Quorum still run as separate processes in same container. Any feedback on [this](https://github.com/chainstack/quorum-docker/pull/11#issuecomment-508627799) is welcome!

File: start_quorum_node.sh [hunk 2]
```diff
@@ -15,25 +15,52 @@ form_consensus_params()
 		CONSENSUS_ARGS="--raft --raftport $RAFTPORT --raftblocktime $RAFTBLOCKTIME"
 		echo "Creating consensus arguments for raft"
 
-        # Raft new node arguments
-        if [[ -n "$RAFTJOINEXISTING" ]]; then
-            CONSENSUS_ARGS="$CONSENSUS_ARGS --raftjoinexisting $RAFTJOINEXISTING"
-            echo "Adding consensus arguments for raft new node"
+		# Raft new node arguments
```
Comment (ToxicWar) [inline start_quorum_node.sh]: Please return spaces instead of tabs.

File: Dockerfile [hunk 3]
```diff
@@ -114,11 +123,18 @@ RUN apt-get update -qq \
         netbase \
     && apt-get clean
 
+# Dependencies are for tessera
+RUN apt-get install -y openjdk-8-jdk && \
```
Comment (ToxicWar) [inline Dockerfile]: Avoid additional packages by specifying `--no-install-recommends`.

**Comment by hayorov** [comment]
Archived due to change deployment. For future implementation.

**Pull request closed without merging**